src/libasr/codegen/llvm_utils.cpp

```diff
@@ -373,6 +373,129 @@ namespace LFortran {
         shift_end_point_by_one(list);
     }
 
+    llvm::Value* LLVMList::find_item_position(llvm::Value* list,
+                                    llvm::Value* item, llvm::Type* el_type) {
+        llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
+                                        get_pointer_to_current_end_point(list));
+        llvm::AllocaInst *i = builder->CreateAlloca(el_type, nullptr);
+        LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), i);
+        llvm::Value* tmp = nullptr;
+
+        /* Equivalent in C++:
+         * int i = 0;
+         * while(list[i] != item && end_point > i) {
+         *     i++;
+         * }
+         *
+         * if (i == end_point) {
+         *    std::cout << "Element: `item` not found in the list";
+         * }
+         */
+
+        llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+        llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+        llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+
+        // head
+        llvm_utils->start_new_block(loophead);
+        {
+            llvm::Value *cond = builder->CreateAnd(builder->CreateICmpNE(
+                read_item(list, LLVM::CreateLoad(*builder, i), false), item),
+                builder->CreateICmpSGT(current_end_point,
+                                       LLVM::CreateLoad(*builder, i)));
+            builder->CreateCondBr(cond, loopbody, loopend);
+        }
+
+        // body
+        llvm_utils->start_new_block(loopbody);
+        {
+            tmp = builder->CreateAdd(
+                        LLVM::CreateLoad(*builder, i),
+                        llvm::ConstantInt::get(context, llvm::APInt(32, 1)));
+            LLVM::CreateStore(*builder, tmp, i);
+        }
+        builder->CreateBr(loophead);
+
+        // end
+        llvm_utils->start_new_block(loopend);
+
+        llvm::Value* cond = builder->CreateICmpEQ(
+                              LLVM::CreateLoad(*builder, i), current_end_point);
+        llvm::Function *fn = builder->GetInsertBlock()->getParent();
+        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
+        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
+        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+        builder->CreateCondBr(cond, thenBB, elseBB);
+
+        builder->SetInsertPoint(thenBB);
+            // "Element: `item` not found in the list"
+            // replace this with Diagnostic Error message.
+            // Print the `item` in the error message.
+        builder->CreateBr(mergeBB);
+
+        llvm_utils->start_new_block(elseBB);
+        llvm_utils->start_new_block(mergeBB);
```

Comment:
```c++
         if (i == end_point) {
            std::cout << "Element: `item` not found in the list";
         }
```
Is this how we implement the above If statement?

I need help moving `print_util` to use here as well.
Also, How do I use `Diagnostic message` here?
@certik @czgdp1807

Reply:
This happens at runtime in the generated code, while the `Diagnostic` class is in LFortran at compile time, so the two live in different worlds at different times.

In here if the item cannot be found we simply do whatever the semantics of the operation is. That is, we have to choose what happens if `ListRemove` can't find the item. I would do what we do for if `ArrayItem` is out of bounds --- we should check it (optionally) and stop the program (with a nice stacktrace eventually).